tests/functional/test_actions.py

```diff
@@ -0,0 +1,127 @@
+import os
```

Comment:
I added lines 34-91, and 121-127. The rest were just moved from it's previous location.